app/models/query_execution.rb

```diff
@@ -13,35 +13,32 @@ def self.perform(result_id, role)
     body = result.compiled_body
     result.mark_running!
     sample_callback = ->(sample) { result.mark_processing_from_sample(sample) }
-    result_csv_generator = ResultCsvGenerator.new(result_id, result.headers)
-    begin
-      Timeout::timeout(QUERY_TIMEOUT) do
-        connection = RedshiftConnectionPool.instance.get(role)
-
-        connection.reconnect_on_failure do
-          query_stream = PgStream::Stream.new(connection.pg_connection, body)
-          result.headers = query_stream.headers
-          result.save!
-
-          rrrc = result.redis_result_row_count
-
-          stream_processor = PgStream::Processor.new(query_stream)
-          stream_processor.register(result_csv_generator.callbacks)
-          stream_processor.register(SampleSkimmer.new(NUM_SAMPLE_ROWS, &sample_callback).callbacks)
-          stream_processor.register(CountPublisher.new(rrrc).callbacks)
-
-          row_count = stream_processor.execute
-          result.mark_complete_with_count(row_count)
-        end
-      end
-    rescue *RedshiftPG::USER_ERROR_CLASSES => e
+
+    connection = RedshiftConnectionPool.instance.get(role)
+
+    connection.reconnect_on_failure do
+      query_stream = PgStream::Stream.new(connection.pg_connection, body)
+      result.headers = query_stream.headers
+      result.save!
+
+      rrrc = result.redis_result_row_count
+
+      stream_processor = PgStream::Processor.new(query_stream)
+      stream_processor.register(ResultCsvGenerator.new(result_id, result.headers).callbacks)
+      stream_processor.register(SampleSkimmer.new(NUM_SAMPLE_ROWS, &sample_callback).callbacks)
+      stream_processor.register(CountPublisher.new(rrrc).callbacks)
+
+      row_count = stream_processor.execute
+      result.mark_complete_with_count(row_count)
+    end
+  rescue *RedshiftPG::USER_ERROR_CLASSES => e
+    csv_service.clear_tmp_file
+    result.mark_failed!(e.message)
+  rescue => e
+    if result && csv_service
+      csv_service.clear_tmp_file
       result.mark_failed!(e.message)
-    rescue Timeout::Error => e
-      File.delete(result_csv_generator.filepath) if File.exist?(result_csv_generator.filepath)
-      result.mark_failed!("Your query timed out because it run for longer than #{QUERY_TIMEOUT} seconds.")
```

Comment:
maybe we should keep some logging when query run longer?

Reply:
well -- we took this line out